### ISSUE TITLE
Auto labeling cluster resources

### ISSUE THREAD

**Issue by fabriziopandini** [kubernetes-sigs/cluster-api#1489]
As an operator I would like to have a consistent set of labels for searching cluster resources.

**Detailed Description**

As the cluster API specification evolves, cluster specification is composed of a growing set of resources e.g. Cluster, AWSCluster, Machine, AWS machine, KubeConfigr, Certificate secrets, Kubeconfig secret etc.

By ensuring everything is part of a cluster consistently labeled it will be possible to simplify the management of a cluster definition and easily implement higher-level tools for deleting/pivoting a cluster

***Goals***
 - Apply a consistent set of labels to all the resources that are part of a cluster (including core objects, provider-specific objects and Kubernetes standard objects like secrets)

***Non-Goals*** 
 - Apply a consistent set of labels to the cluster API components itself (the manager, CRD definitions etc.)

/kind feature

**Comment by prankul88** [comment]
@fabriziopandini I can work on this if you like.

**Comment by fabriziopandini** [comment]
@prankul88 great! let's wait for some more feedback/issue triage before starting though

**Comment by fabriziopandini** [comment]
In order to simplify for deleting/pivoting, on top of auto labeling, it is necessary also a generic convention for navigating objects and this requires more thinking
So I propose to decrease priority or to close and eventually re-open in future

**Comment by ncdc** [comment]
Done

**Comment by fabriziopandini** [comment]
@vincepri considering that clusterctl now relies on the OwnerRef chain (see https://master.cluster-api.sigs.k8s.io/clusterctl/provider-contract.html#ownerreferences-chain), I propose to close this issue.
WDYT?
